server/index.js

```diff
@@ -61,7 +63,21 @@ export default class Server {
       availableChunks: dev ? {} : getAvailableChunks(this.dir, this.dist)
     }
 
-    this.setAssetPrefix(this.config.assetPrefix)
+    // Allow configuration from next.config.js to be passed to the server / client
+    if (this.nextConfig.config) {
+      // Initialize next/config with the environment configuration
+      envConfig.setConfig(this.nextConfig.config)
+
+      // Only the `public` key is exposed to the client side
+      // It'll be rendered as part of __NEXT_DATA__ on the client side
+      if (this.nextConfig.config.public) {
```

Comment:
Added here: https://github.com/zeit/next.js/pull/3882/commits/859c38a6efe955c2423a39f8297fa7afcf0f54e9

Because I don't want to add it to getInitialProps if it's not configured.

Reply:
@timneutkens hmm. Okay that's not what I wanted. But I feel like this is just fine.